src/java.base/share/classes/java/lang/invoke/MethodHandleProxies.java

```diff
@@ -274,32 +274,26 @@ public static MethodHandle wrapperInstanceTarget(Object x) {
     }
 
     private static boolean isObjectMethod(Method m) {
-        switch (m.getName()) {
-        case "toString":
-            return (m.getReturnType() == String.class
-                    && m.getParameterCount() == 0);
-        case "hashCode":
-            return (m.getReturnType() == int.class
-                    && m.getParameterCount() == 0);
-        case "equals":
-            return (m.getReturnType() == boolean.class
-                    && m.getParameterCount() == 1
-                    && m.getParameterTypes()[0] == Object.class);
-        }
-        return false;
+        return switch (m.getName()) {
+            case "toString" -> (m.getReturnType() == String.class
```

Comment:
the extra parenthesis are not needed

Reply:
Parenthesis removed, as suggested. See a8706b0